src/components/templates/PartyMap/components/RoomModal/RoomModal.tsx

```diff
@@ -61,49 +65,40 @@ export const RoomModal: React.FC<RoomModalProps> = ({
 export interface RoomModalContentProps {
   room: Room;
   venueName: string;
+  venueEvents?: WithVenueId<WithId<VenueEvent>>[];
 }
 
 export const RoomModalContent: React.FC<RoomModalContentProps> = ({
   room,
   venueName,
+  venueEvents,
 }) => {
-  const venueEvents = useSelector(venueEventsSelector);
-
   const { enterRoom, recentRoomUsers } = useRoom({ room, venueName });
 
   const [enterRoomWithSound] = useCustomSound(room.enterSound, {
     interrupt: true,
     onend: enterRoom,
   });
 
-  const roomEvents = useMemo(() => {
+  const renderedRoomEvents = useMemo(() => {
     if (!venueEvents) return [];
 
-    return venueEvents.filter(
-      (event) =>
-        event.room === room.title && isBefore(Date.now(), eventEndTime(event))
-    );
-  }, [room, venueEvents]);
-
-  const currentEvent = getCurrentEvent(roomEvents);
-
-  const renderedRoomEvents = useMemo(
-    () =>
-      roomEvents.map((event, index: number) => (
-        <ScheduleItem
-          // @debt Ideally event.id would always be a unique identifier, but our types suggest it
-          //   can be undefined. Because we can't use index as a key by itself (as that is unstable
-          //   and causes rendering issues, we construct a key that, while not guaranteed to be unique,
-          //   is far less likely to clash
-          key={event.id ?? `${event.room}-${event.name}-${index}`}
-          event={event}
-          isCurrentEvent={currentEvent && event.name === currentEvent.name}
-          onRoomEnter={enterRoomWithSound}
-          roomUrl={room.url}
-        />
-      )),
-    [currentEvent, enterRoomWithSound, room.url, roomEvents]
-  );
+    const currentEvent = getCurrentEvent(venueEvents);
+
+    return venueEvents.map((event, index: number) => (
+      <ScheduleItem
+        // @debt Ideally event.id would always be a unique identifier, but our types suggest it
+        //   can be undefined. Because we can't use index as a key by itself (as that is unstable
+        //   and causes rendering issues, we construct a key that, while not guaranteed to be unique,
+        //   is far less likely to clash
+        key={event.id ?? `${event.room}-${event.name}-${index}`}
+        event={event}
+        isCurrentEvent={currentEvent && event.name === currentEvent.name}
+        onRoomEnter={enterRoomWithSound}
+        roomUrl={room.url}
+      />
+    ));
+  }, [enterRoomWithSound, room.url, venueEvents]);
```

Comment:
Since `getCurrentEvent` already processes the array with `.find`, and then we are processing the array again below with `.map`, this is inefficient.

Looking at the definition of `getCurrentEvent`:

- https://github.com/sparkletown/sparkle/blob/staging/src/utils/event.ts#L18-L26

The main bit of work is:

```
(event) =>
      event.start_utc_seconds < currentTimeInUTCSeconds &&
      event.start_utc_seconds + event.duration_minutes > currentTimeInUTCSeconds
```

In english:
- is the event's start before the current time
- does the event end after the current time

`isEventLive` uses `getEventInterval` to get the start/end time of the event, then uses `isWithinInterval` to check if the current time is within that.

- https://date-fns.org/v2.21.3/docs/isWithinInterval

So basically, `isEventLive` does the equivalent to this code.

---

Based on the above, we should change `getCurrentEvent` to be as follows:

```typescript
export const getCurrentEvent = (roomEvents: VenueEvent[]) => roomEvents.find(isEventLive);
```

And then there is no need to use `getCurrentEvent` here, because we can just use `isEventLive` directly within the `.map`:

```suggestion
    return venueEvents.map((event, index: number) => (
      <ScheduleItem
        // @debt Ideally event.id would always be a unique identifier, but our types suggest it
        //   can be undefined. Because we can't use index as a key by itself (as that is unstable
        //   and causes rendering issues, we construct a key that, while not guaranteed to be unique,
        //   is far less likely to clash
        key={event.id ?? `${event.room}-${event.name}-${index}`}
        event={event}
        isCurrentEvent={isEventLive(event)}
        onRoomEnter={enterRoomWithSound}
        roomUrl={room.url}
      />
    ));
  }, [enterRoomWithSound, room.url, venueEvents]);
```

Reply:
Fixed in https://github.com/sparkletown/sparkle/pull/1538/commits/7eba21f8e2fa302d88d03ab53d4809bd15fc2449 + https://github.com/sparkletown/sparkle/pull/1538/commits/ff2350207191e67a797f79326ad10ed16dea66bf + https://github.com/sparkletown/sparkle/pull/1538/commits/f901ee13b588fafddff3985c612f818ae78a34a2 + https://github.com/sparkletown/sparkle/pull/1538/commits/3597fb1cb5fa80d044f763613e9b568055366e8c